traitsui/editors/list_editor.py

```diff
@@ -84,6 +86,18 @@ class ListEditor(EditorFactory):
     #: Show a right-click context menu for the notebook tabs?  (Qt only)
     show_notebook_menu = Bool(False)
 
+    #: Factory callable that will be called to create the new element to add to
+    #: this list. If None, the default value for the trait of interest is used.
+    default_factory = Callable()
+
+    #: List of positional arguments to be passed to the default factory
+    #: callable when creating new elements
+    default_args = List()
+
+    #: Dictionary of keyword arguments to be passed to the default factory
+    #: callable when creating new elements
+    default_kwargs = Dict()
```

Comment:
i'd rename `default_args` and `default_kwargs` to `item_factory_args` and `item_factory_kwargs` to reflect where they get used.

I think this will make the calling code readable - 

```
value = self.factory.item_factory(*self.factory.item_factory_args, **self.factory.item_factory_kwargs)
```